src/types.ts

```diff
@@ -1,7 +1,82 @@
+import * as pulumi from '@pulumi/pulumi';
+import { Stack, StackProps } from 'aws-cdk-lib/core';
+import { ResourceMapping } from './interop';
+/**
+ * Options specific to the Stack component.
+ */
+export interface StackOptions extends pulumi.ComponentResourceOptions {
+    /**
+     * Specify the CDK Stack properties to asociate with the stack.
+     */
+    props?: StackProps;
+
+    /**
+     * Defines a mapping to override and/or provide an implementation for a CloudFormation resource
+     * type that is not (yet) implemented in the AWS Cloud Control API (and thus not yet available in
+     * the Pulumi AWS Native provider). Pulumi code can override this method to provide a custom mapping
+     * of CloudFormation elements and their properties into Pulumi CustomResources, commonly by using the
+     * AWS Classic provider to implement the missing resource.
+     *
+     * @param logicalId The logical ID of the resource being mapped.
+     * @param typeName The CloudFormation type name of the resource being mapped.
+     * @param props The bag of input properties to the CloudFormation resource being mapped.
+     * @param options The set of Pulumi ResourceOptions to apply to the resource being mapped.
+     * @returns An object containing one or more logical IDs mapped to Pulumi resources that must be
+     * created to implement the mapped CloudFormation resource, or else undefined if no mapping is
+     * implemented.
+     */
+    remapCloudControlResource?(
+        logicalId: string,
+        typeName: string,
+        props: any,
+        options: pulumi.ResourceOptions,
+    ): ResourceMapping | undefined;
+}
+
 /**
  * The pulumi provider to read the schema from
  */
 export enum PulumiProvider {
     // We currently only support aws-native provider resources
     AWS_NATIVE = 'aws-native',
 }
+
+export abstract class IStackComponent extends pulumi.ComponentResource {
```

Comment:
Is IFoo convention for TypeScript too, sounds like a .NET convention for interfaces to me?

Reply:
Is this abstract class suggested for users to fill out?

Reply:
I'll add some comments here. It is not for users to use, just us internally. I'll also change the name, the `IFoo` is a CDK convention, but only for interfaces (which this one started out as before I changed it to an abstract class)

Reply:
Changed the naming here and added some comments.